addon/index.js

```diff
@@ -186,6 +188,8 @@ function installExperiment(experiment) {
 }
 
 function uninstallSelf() {
+  // HACK: onUnload often doesn't seem to get called, here.
+  exports.onUnload('uninstall');
```

Comment:
In what case is it not called? When I log the `onUnload` on line 360 I always see it twice, when I remove this line I see it once, albeit not until after this function has completed.
<img width="653" alt="screen shot 2016-09-28 at 10 43 42 am" src="https://cloud.githubusercontent.com/assets/87619/18925434/d94b7f30-8568-11e6-84f4-019fb66b9b22.png">